### ISSUE TITLE
NullReference when calling Bass.GetDeviceInfo()

### ISSUE THREAD

**Issue by trinitrot0luene** [ppy/osu#2666]
(Note: I'm not sure whether it would be best to declare the issue here or in the framework repo, so if it'd be better to move it there, let me know!)

Logs and debug information:
- [debugger object inspection](https://anime.is-bad.com/76c939.png)
- [full log output](https://hastebin.com/zuhezavoru.log)
- [native stacktrace only](https://hastebin.com/duqahupico.txt)

In [the framework AudioManager](https://github.com/ppy/osu-framework/blob/84fdfc77a86d581638e69f5e8061c118de4b30f9/osu.Framework/Audio/AudioManager.cs#L186)
```cs
private List<DeviceInfo> getAllDevices()
{
    int deviceCount = Bass.DeviceCount; // DeviceCount is null
    List<DeviceInfo> info = new List<DeviceInfo>();
    for (int i = 0; i < deviceCount; i++)
        info.Add(Bass.GetDeviceInfo(i));

    return info;
}
```

I get the same error both in Release and Debug config in MonoDevelop, and when running from console.

I've tried swapping audio devices around to see if maybe things aren't being picked up but I don't have any known audio problems on this system.

Not sure how to further debug this.

**Comment by peppy** [comment]
I'm guessing you're on linux?

**Comment by trinitrot0luene** [comment]
Yep.

Specs:
```
OS: Ubuntu 17.10 artful
Kernel: x86_64 Linux 4.13.0-43-generic
CPU: Intel Core i7-3770K @ 8x 3.9GHz [27.8°C]
GPU: GeForce GTX 1060 6GB
RAM: 2405MiB / 7908MiB
```
Output of `arecord -L`
- [console output](https://hastebin.com/onomojuded.txt)

**Comment by peppy** [comment]
Possibly related to ppy/osu-framework#1540.

**Comment by SlowLogicBoy** [comment]
It is the same as https://github.com/ppy/osu-framework/issues/1540 after https://github.com/ppy/osu-framework/pull/1617 it got fixed.

**Comment by peppy** [comment]
Going forward we're not supporting mono officially, so a long as it's working correctly under netcore things should be fine. Closing as resolved by #1617.
